infra/growth/src/resources/rules.js

```diff
@@ -827,9 +827,21 @@ class SocialShareRule extends SingleEventRule {
       // No user passed. Return zero.
       return reward
     }
+
+    // Load any proxy associated with the wallet address.
```

Comment:
@shahthepro This should not hurt but theoretically it should not be needed since identities are stored in the DB by owner address. See the [listener code](https://github.com/OriginProtocol/origin/blob/master/infra/discovery/src/listener/handler_identity.js#L273).

And then the ethAddress passed to all rules methods should be the owner address as well since it comes from the auth token signed by the user, which has to be the wallet address (proxy can't sign).

Do you have an example of an address where identity was keyed by proxy and that required this fix ?

Reply:
@franckc 

It happens in local for me. I updated the identity of `0xf17...` changing the name from "Stan James" to "Another World". It created another entry with the proxy of `0xf17...` (`0xba...`)

![Screenshot 2019-07-31 at 9 21 52 PM](https://user-images.githubusercontent.com/10547529/62227319-4ea19c00-b3d9-11e9-9479-7f0ffe97f6c7.png)